rules/CodeQuality/Rector/FunctionLike/SimplifyUselessVariableRector.php

```diff
@@ -134,7 +132,8 @@ private function shouldSkipStmt(Return_ $return, Stmt $previousStmt): bool
             return true;
         }
 
-        if ($this->returnAnalyzer->hasByRefReturn($return)) {
+        $functionReflection = $scope->getFunction();
+        if ($functionReflection instanceof FunctionReflection && $functionReflection->returnsByReference()->yes()) {
```

Comment:
maybe I am missing something, but does this check only make sense for a `CallLike` node?
If so it contains the name of the function/method called then?

Does take it from scope make sense in this case?

Reply:
custom Node visitor probably needed, like `parentStmtTypes` collect types of parent, but with on every enter stmt on each `FunctionLike` so then it can be grabbed via:

```php
$node->getAttribute('functionLike');
```

Reply:
@TomasVotruba for less memory usage, attribute may not be an object node, but a "is by reference" flag, eg `true` or `false`, so it need to fill the stack return with same type to have "is by reference" flag when found. That is the possible solution I can imagine.

Reply:
@TomasVotruba I will try create new Visitor for it 👍

Reply:
@TomasVotruba I created PR https://github.com/rectorphp/rector-src/pull/3826 for it.

Reply:
Perfect!